packages/x-data-grid/src/hooks/core/useGridIsRtl.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { useRtl } from '@mui/system/RtlProvider';
+import { GridPrivateApiCommon } from '../../models/api/gridApiCommon';
+
+export const useGridIsRtl = (apiRef: React.MutableRefObject<GridPrivateApiCommon>): void => {
+  const isRtl = useRtl();
+
+  if (apiRef.current.state.isRtl !== undefined) {
+    apiRef.current.state.isRtl = isRtl;
+  }
```

Comment:
Shouldn't this be `=== undefined`?

Reply:
Whoops, it should, indeed. 🙈 
```suggestion
  if (apiRef.current.state.isRtl === undefined) {
    apiRef.current.state.isRtl = isRtl;
  }
```